.github/workflows/multi-arch-build.yaml

```diff
@@ -0,0 +1,119 @@
+          platforms: ${{ env.PLATFORMS }}
+          push: true
+          tags: |
+            localhost:5000/buildah/${{ matrix.source }}
+
+      # Simple verification that container works + get version for public image tag
+      - name: amd64 sniff test
+        id: run
+        env:
+          MATRIX_VERSION: ${{ matrix.source }}
+        run: |
+          # image tag is master for upstream source, actual version of buildah binary in the container for stable source
+          VERSION=v"$(docker run localhost:5000/buildah/${MATRIX_VERSION} buildah version | head -n 1 | awk '{print $2}')"
+          if [ "${MATRIX_VERSION}" == "upstream" ]; then
+            VERSION=master
+          fi
+          echo ::set-output name=version::$VERSION
+          echo ::set-output name=created::$(date -u --iso-8601=seconds)
+
+      # Push to buildah Quay repo stable and upstream Buildah (from matrix.source)
+      - name: Login to buildah Quay registry
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.BUILDAH_QUAY_REGISTRY}}
+          username: ${{ secrets.BUILDAH_QUAY_USERNAME }}
+          password: ${{ secrets.BUILDAH_QUAY_PASSWORD }}
+
+      - name: Push images to buildah Quay
+        uses: docker/build-push-action@v2
+        with:
+          cache-from: type=registry,ref=localhost:5000/buildah/${{ matrix.source }}
+          cache-to: type=inline
+          context: contrib/buildahimage/${{ matrix.source }}
+          file: ./contrib/buildahimage/${{ matrix.source }}/Dockerfile
+          platforms: ${{ env.PLATFORMS }}
+          push: true
+          tags: |
+            ${{ env.BUILDAH_QUAY_REGISTRY}}/${{ matrix.source }}:${{ steps.run.outputs.version }}
+          labels: |
+            org.opencontainers.image.source=${{ github.event.repository.html_url }}
+            org.opencontainers.image.revision=${{ github.sha }}
+            org.opencontainers.image.created=${{ steps.run.outputs.created }}
+
+      # Push to containers Quay repo only stable Buildah
+      - name: Login to containers Quay registry
+        if: ${{ matrix.source == 'stable' }}
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.CONTAINERS_QUAY_REGISTRY}}
+          username: ${{ secrets.CONTAINERS_QUAY_USERNAME }}
+          password: ${{ secrets.CONTAINERS_QUAY_PASSWORD }}
+
+      - name: Push images to containers Quay
+        if: ${{ matrix.source == 'stable' }}
+        uses: docker/build-push-action@v2
+        with:
+          cache-from: type=registry,ref=localhost:5000/buildah/${{ matrix.source }}
+          cache-to: type=inline
+          context: contrib/buildahimage/${{ matrix.source }}
+          file: ./contrib/buildahimage/${{ matrix.source }}/Dockerfile
+          platforms: ${{ env.PLATFORMS }}
+          push: true
+          tags: |
+            ${{ env.CONTAINERS_QUAY_REGISTRY}}/${{ matrix.source }}:${{ steps.run.outputs.version }}
```

Comment:
This doesn't seem correct, maybe?  There are multiple images at quay.io/containers: "podman", "buildah", and "skopeo".  I think this would push up a "quay.io/containers/stable" which is an incongruity.  Maybe should this be:

`${{ env.CONTAINERS_QUAY_REGISTRY}}/buildah:${{ matrix.source }}`

?

Reply:
Oof, nvm I was reading `$CONTAINER_QUAY_REGISTRY` wrongly.

Reply:
Err, double-oof, no, I think this is a typo.  It should be `${{ env.CONTAINERS_QUAY_REGISTRY}}/buildah:${{ matrix.source }}` (or similar).

Reply:
fixed names and tags (+ added desired output as comment)